packages/editor/src/browser/editor-keybinding-contexts.ts

```diff
@@ -23,7 +23,8 @@ export class EditorTextFocusContext implements KeybindingContext {
     protected readonly editorManager: EditorManager;
 
     isEnabled(): boolean {
-        return !!this.getEditor();
+        const editor = this.getEditor();
+        return !!editor && !editor.editor.isSuggestVisible();
```

Comment:
We should consider to limit the scope of this change to the git blame by introducing the special keybinding context for the clean git annotation command or to introduce another context in the editor checking that there is no other widgets.

Generally it is true that the text editor has focus even if the suggest or other widgets are visible. Changing it could break some commands. You can observe it by trying enable git annotations with the suggest widget and without. It behaves differently. It does not harm so much in this case, but if there is a command enabling another editor widget (think about find references) then it won't work.

Could you please also check that it is not case for other widgets. It seems to be that find and rename widgets are affected as well, based on https://github.com/eamodio/vscode-gitlens/blob/57226d54d1e929be04b02ee31ca294c50305481b/package.json#L2857.

(minor) I've introduced `canHandle` for checking, that one does not need to change `isEnabled`.

Reply:
> Separate contexts for the text editor focus with widget and without

That was my [initial plan](https://github.com/theia-ide/theia/issues/1688#issuecomment-381175928) then I thought it adds unnecessary complexity. Let's split it into two separate contexts.

> Could you please also check

Sure.

> it is too much to assume and ask from editor implementors

I agree.

> rebind it in the monaco extension

I like it!

> minor

👍

Reply:
> Could you please also check that it is not case for other widgets.

It is. For instance, I cannot hide the search and replace widget with `Esc`.